Structure_Engine/Modify/OrientTowards.cs

```diff
@@ -0,0 +1,97 @@
+        /***************************************************/
+
+        [Description("Orients the normal of the Bar towards the provided point. The normal is calculated as the vector from the centre of the bar to the provided point.")]
+        [Input("bar", "The Bar to update the orientation of.")]
+        [Input("orientationPoint", "The point to orient the normal of the Bar towards.")]
+        [Output("bar", "Bar with updated orientation. If the calcualtion of the orientation angle fails, the unmodified bar is returned.")]
+        public static Bar OrientTowards(this Bar bar, Point orientationPoint)
+        {
+            Point centre = (bar.StartNode.Position + bar.EndNode.Position) / 2;
+            Vector normal = (orientationPoint - centre).Normalise();
+
+            return bar.SetNormal(normal);
+        }
+
+        /***************************************************/
+
+        [Description("Orients the direction of the local x of the Panel towards the provided point. The local x is calculated as the vector from the centroid of the Panel to the provided point.")]
+        [Input("panel", "The Panel to update the local orientation of.")]
+        [Input("orientationPoint", "The point to orient the local x of the Panel towards.")]
+        [Output("panel", "Panel with updated local orientation. If the calcualtion of the orientation angle fails, the unmodified Panel is returned.")]
+        public static Panel OrientTowards(this Panel panel, Point orientationPoint)
+        {
+            Point centre = panel.Centroid();
+            Vector localX = (orientationPoint - centre).Normalise();
+
+            return panel.SetLocalOrientation(localX);
+        }
+
+        /***************************************************/
+
+        [Description("Orients the direction of the local x of each FEMeshFace of the FEMesh towards the provided point. The local x vectors is calculated as the vector from the centroid of each FEMeshFace of the FEMesh to the provided point.")]
+        [Input("mesh", "The FEMesh to update the local orientations of.")]
+        [Input("orientationPoint", "The point to orient the local x vectors of the FEMeshFaces of the FEMesh towards.")]
+        [Output("mesh", "FEMesh with updated local orientations. If the calcualtion of the orientation angle fails for an FEMeshFace, it remains unchanged.")]
+        public static FEMesh OrientTowards(this FEMesh mesh, Point orientationPoint)
+        {
+            FEMesh clone = mesh.GetShallowClone() as FEMesh;
+            clone.Faces = clone.Faces.Select(x => x.OrientTowards(mesh, orientationPoint)).ToList();
+            return clone;
+        }
+
+        /***************************************************/
+
+        [Description("Orients the direction of the local x of the FEMeshFace towards the provided point. The local x vectors is calculated as the vector from the centroid of the FEMeshFace of to the provided point.")]
```

Comment:
local x vectors are calculated

Reply:
keeping 'is', but getting rid of s as single face